pushnotifications/src/main/java/com/pusher/pushnotifications/reporting/ReportingJobService.kt

```diff
@@ -63,12 +63,21 @@ class ReportingJobService: JobService() {
       return b
     }
 
-    fun fromBundle(bundle: Bundle): ReportEvent {
-      val event = ReportEventType.valueOf(bundle.getString(BUNDLE_EVENT_TYPE_KEY))
-      when (event) {
+    fun fromBundle(bundle: Bundle): ReportEvent? {
+      val instanceId = bundle.getString(BUNDLE_INSTANCE_ID_KEY)
+      if (instanceId == null) {
+        // it's possible that we are processing a bundle that was created with an old SDK
+        // version that didn't had this key. Our migration strategy is to drop the reporting
+        // as it's (a) a rare one-time transition and (b) it's a best effort feature.
+        return null
```

Comment:
I think the only change to work around the missing migration should be throwing a specific exception here and catching only that specific exception when calling this method (narrow try-catch).